tests/projects/c++/modules/stdmodules/xmake.lua

```diff
@@ -1,14 +1,11 @@
 add_rules("mode.debug", "mode.release")
-
-add_cxxflags("clang::-stdlib=libc++")
-
```

Comment:
It still does not work for me.

```
/usr/bin/clang -c -x c++-module --precompile -Qunused-arguments -m64 -fvisibility=hidden -fvisibility-inlines-hidden -O3 -std=c++2b -fmodules -fbuiltin-module-map -fimplicit-modules -DNDEBUG -fmodules-cache-path=build/.gens/mod/linux/x86_64/release/rules/modules/cache -o build/.gens/mod/linux/x86_64/release/rules/modules/cache/my_module.pcm src/my_module.mpp
/usr/bin/clang -c -Qunused-arguments -m64 -fvisibility=hidden -fvisibility-inlines-hidden -O3 -std=c++2b -fmodules -fbuiltin-module-map -fimplicit-modules -DNDEBUG -fmodules-cache-path=build/.gens/mod/linux/x86_64/release/rules/modules/cache -o build/.objs/mod/linux/x86_64/release/src/my_module.mpp.o build/.gens/mod/linux/x86_64/release/rules/modules/cache/my_module.pcm
checking for flags (-MMD -MF) ... ok
> clang "-MMD" "-MF" "/dev/null"
checking for flags (-fdiagnostics-color=always) ... ok
> clang "-fdiagnostics-color=always"
error: @programdir/modules/private/async/runjobs.lua:256: @programdir/rules/c++/modules/modules_support/clang.lua:194: @programdir/modules/core/tools/gcc.lua:709: src/my_module.mpp:3:8: fatal error: module 'std' not found
import std;
~~~~~~~^~~
1 error generated.
```

when I revert ` --add_cxxflags("clang::-stdlib=libc++")` in xmake.lua, it works.

Reply:
if you build it manualy you need to add --cxxflags="-stdlib=libc++" to the xmake f command, i removed the flag in the xmake.lua because the test should be compatible with both libstdc++ and libc++

(check test_stdmodules.lua in the test suite)

Reply:
It should be automatically supported, or at least not left to the user to configure it manually. This will lead to many users encountering build errors and thinking that the current clang version doesn't support std modules, and they won't take the time to work around it or set `--cxxflags="--stdlib=libc++"` to make it work.

Also, on my machine, only libc++ clang supports std modules, while the default libstdc++ does not. Therefore, if I switch to the clang toolchain, stdmodules always don't work.

Reply:
>>> It should be automatically supported, or at least not left to the user to configure it manually. This will lead to many users encountering build errors and thinking that the current clang version doesn't support std modules, and they won't take the time to work around it or set --cxxflags="--stdlib=libc++" to make it work.

maybe i can assert with a proper error message if import std is detected but --stdlib=libc++ is not set ?

>>> Also, on my machine, only libc++ supports std modules, while the default libstdc++ does not. Therefore, if I switch to the clang toolchain, stdmodules always don't work.

libstdc++ don't support c++23 std modules on any platform / compiler ^^

Reply:
> libstdc++ don't support c++23 std modules on any platform / compiler ^^

so we should always set `add_cxxflags("clang::-stdlib=libc++")` for std modules / clang.

Reply:
like

```
xmake b                                                            17:26:04 
[  0%]: generating.module.deps test/test.cpp
error: On llvm <= 16 standard C++ modules are not supported, they can be emulated through clang modules and supported only on libc++, please add -stdlib=libc++ cxx flag
```

Reply:
it should work now